scipy/linalg/tests/test_basic.py

```diff
@@ -1043,6 +1114,14 @@ def test_empty_edge_cases(self):
         with assert_raises(ValueError, match='Last 2 dimensions'):
             det(np.array([[[]]]))
 
+    @pytest.mark.xfail(reason="det(float32) -> float64, is it a bug?")
```

Comment:
This looks like it's on purpose:
```python
    # ... avoid overflows for single precision
    # Cf. np.linalg.det(np.diag([1e+38, 1e+38]).astype(np.float32))
    dtype_char = a1.dtype.char
    if dtype_char in 'fF':
        dtype_char = 'd' if dtype_char.islower() else 'D'
```

The documented return type is `float or complex` - not super clear, but it's on purpose. `det` upcasting to 64-bit isn't unreasonable I think. So I suggest improving the docs a bit and updating this test so it passes.

Reply:
Good catch, thanks! 
Indeed, it's even documented, https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.det.html

>    Even the input array is single precision (float32 or complex64), the result
    will be returned in double precision (float64 or complex128) to prevent
    overflows.

PR updated to make it consistent for empty and non-empty arrays.